src/orb-release/cli.js

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-disable no-param-reassign */
+
+const diff = require('diff');
+const fs = require('fs');
+const gitLogParser = require('git-log-parser');
+const getStream = require('get-stream');
+const shell = require('shelljs');
+const yargs = require('yargs');
+
+const releaseTypes = [
+  '',
+  'patch',
+  'minor',
+  'major',
+];
+const logger = console; // executed in CircleCI runtime
+let config = {};
+
+function getReleaseDate() {
+  if (fs.existsSync(config.changelog)) {
+    const cl = fs.readFileSync(config.changelog).toString('utf-8');
```

Comment:
Filesystem sync operations should be avoided. You should rather do `await fs.exists(...)` and `const cl = await fs.readFile(...)` (+ cascading async functions).

Reply:
Ok. But why?

Reply:
Sync operations will block the main thread from performing any other operations. They should be avoided in general.

Reply:
Well, in this case there is nothing else which needs to be going on 😏

Reply:
I looked into using async `fs` functions and quickly ran into several problems... mainly related to the fact that the async API methods usually don't return anything, so `await`ing them won't work... I'd have to refactor a lot of code to callback-based, which in this case - and in my opinion - is not really worth the effort: this code will be executed synchronously and only once per circleci job.

Duly noted, though :)

Reply:
https://nodejs.org/api/fs.html#fs_fspromises_readfile_path_options is your friend ;)

Reply:
We do that everywhere ;)
https://github.com/adobe/helix-cli/search?q=readFile%28&unscoped_q=readFile%28